Assets/Scripts/PlayerCharacter.cs

```diff
@@ -57,87 +68,127 @@ public class PlayerCharacter : MonoBehaviour
         {
             if (Input.GetButtonDown("Jump") && jumpCooldown == 0)
             {
                 Jump();
-                //allows player to jump and checks to make sure character can't jump when dead
-            }
-            deathTimer = 0;
-        }
-        if (isInDeath)
-        {
-            anim.SetBool("IsDead", true);
-            if (Time.realtimeSinceStartup - deathTimer >= timeToDeath)
-            {
-                Respawn();
-                //if death timer has gone up and matches time to death, it will enter the next stage of respawn.
-                GetComponent<Rigidbody2D>().velocity = new Vector2(0, 0);
             }
-            GetComponent<Rigidbody2D>().velocity = myRigidBody.velocity/deathVector;
+            DeathTimerZero();
         }
+        
         GetMovementInput();
+        JumpCheck();
+        UpdateAnimationParameters();
+    }
+
+    private void JumpCheck()
+    {
+        ///Checks what jump the character is on and applies and checks cooldown as well as apply x movement reduction while jumping
         if (jumpNumber > 0)
         {
-            speedJumpReducer = 2; 
-            // reduces jump height with each subsequent jump.
+            speedJumpReducer = 2;
+            // Applies x movement reduction while jumping
         }
         if (jumpCooldown > 0)
         {
             jumpCooldown--;
         }
         if (myRigidBody.velocity.y == 0)
         {
-            jumpNumber = 0; 
+            jumpNumber = 0;
             //if the character is not moving vertically, the player has taken at least a moment to rest before jumping again.
         }
         if (jumpNumber == 0)
         {
-            speedJumpReducer = 1; 
+            speedJumpReducer = 1;
             //first jump has no vertical distance reducer.
         }
-        UpdateAnimationParameters();
+    }
+
+    private void DeathTimerZero()
+    {
+        deathTimer = 0;
+    }
+
+    private void DyingUpdate()
+    {
+        //<summary>if death timer has gone up and matches time to death, it will enter the next stage of respawn.</summary>
+        anim.SetBool("IsDead", true);
+        if (deathTimer > timeToDeath)
+        {
+            Respawn();
+            myRigidBody.velocity = new Vector2(0, 0);
+        }
+        DeathVector();
+        deathTimer += Time.deltaTime;
+    }
+
+    private void DeathVector()
+    {
+        //<summary>DeathVector acts to slow the player once they are in the process of dying.</summary>
+        myRigidBody.velocity = myRigidBody.velocity / deathVector;
     }
 
     private void UpdateAnimationParameters()
     {
-        grounded = Physics2D.OverlapCircle(groundCheck.position, groundRadius, whatIsGround);
-        anim.SetBool("IsOnGround", grounded);
-        //sets tells animator if character is falling or on the ground.
-        if (myRigidBody.velocity.y < 0 && grounded == false)
+        IsOnGround();
```

Comment:
Make this method name a verb. e.g. `UpdateIsOnGround`. Name it after what it does.